packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -551,6 +553,17 @@ export class MergeTree {
 	public mergeTreeDeltaCallback?: MergeTreeDeltaCallback;
 	public mergeTreeMaintenanceCallback?: MergeTreeMaintenanceCallback;
 
+	/**
+	 * Cache the sliding destination for segments after a remove to improve the
+	 * pathological case in which we slide the same segment multiple times
```

Comment:
the case isn't very pathological :). it's just removal of multiple segments. If we mark a contiguous range removed, we don't want to repeat traversal work for each segment in the range (as that would scale O(#removed segments^2)).